app/grandchallenge/participants/signals.py

```diff
@@ -15,7 +18,31 @@ def process_registration(
         RegistrationRequest.objects.filter(pk=instance.pk).update(
             status=instance.status
         )
+    elif created and instance.challenge.require_participant_review:
+        action.send(
+            sender=instance.user,
+            verb="requested access to",
+            target=instance.challenge,
+        )
+    if not is_following(instance.user, instance):
```

Comment:
Does the user who made the request get notified of acceptance and rejection? 

An aside: do they get notified for Archive/Algorithm/ReaderStudy acceptance and rejection?

Reply:
If I'm reading your tests correctly I think that they do, which is what we want :-)

Reply:
Yes, the requester gets notified of acceptance / rejection of the request, both for challenges as well as for archives/algorithms/reader studies ;-) That's because they follow the request object, and whenever the status changes, we send an action with the request object as sender and without a target / action object. Actions with only a sender in turn trigger notifications for all followers of the sender.